http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -505,12 +485,247 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
+//        RuntimeServiceInterp.processPersistentDiskRequest(
+//          diskReq,
+//          config.leoKubernetesConfig.diskConfig.defaultZone, // this need to be updated if we support non-default zone for k8s apps
+//          googleProject,
+//          userInfo,
+//          petSA,
+//          appTypeToFormattedByType(req.appType),
+//          authProvider,
+//          config.leoKubernetesConfig.diskConfig
+//        )
+//      )
+
+      saveApp <- F.fromEither(
+        getSavableApp(
+          cloudContext,
+          appName,
+          userInfo.userEmail,
+          samResourceId,
+          req,
+          None, // diskResultOpt.map(_.disk),
+          None, // lastUsedApp,
+          petSA,
+          NodepoolLeoId(123123123), // TODO: Fix this //nodepool.id,
+          ctx
+        )
+      )
+      app <- appQuery.save(saveApp, Some(ctx.traceId)).transaction
+
+      createAppV2Message = CreateAppV2Message(
+        workspaceId,
+        app.id,
+        app.appName,
+        None, // diskResultOpt.flatMap(d => if (d.creationNeeded) Some(d.disk.id) else None),
+        req.customEnvironmentVariables,
+        req.appType,
+        app.appResources.namespace.name,
+        Some(ctx.traceId)
+      )
+      _ <- publisherQueue.offer(createAppV2Message)
+    } yield ()
+
+  override def deleteAppV2(userInfo: UserInfo, workspaceId: WorkspaceId, appName: AppName, deleteDisk: Boolean)(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] = for {
+    ctx <- as.ask
+    appOpt <- KubernetesServiceDbQueries.getActiveFullAppByWorkspaceIdAndAppName(workspaceId, appName).transaction
+    appResult <- F.fromOption(
+      appOpt,
+      AppNotFoundByWorkspaceIdException(workspaceId, appName, ctx.traceId, "No active app found in DB")
+    )
+    listOfPermissions <- authProvider.getActions(appResult.app.samResourceId, userInfo)
+
+    // throw 404 if no GetAppStatus permission
+    hasReadPermission = listOfPermissions.toSet.contains(AppAction.GetAppStatus)
+    _ <-
+      if (hasReadPermission) F.unit
+      else
+        F.raiseError[Unit](
+          AppNotFoundByWorkspaceIdException(workspaceId, appName, ctx.traceId, "no read permission")
+        )
+
+    // throw 403 if no DeleteApp permission
+    hasDeletePermission = listOfPermissions.toSet.contains(AppAction.DeleteApp)
+    _ <- if (hasDeletePermission) F.unit else F.raiseError[Unit](ForbiddenError(userInfo.userEmail))
+
+    canDelete = AppStatus.deletableStatuses.contains(appResult.app.status)
+    _ <-
+      if (canDelete) F.unit
+      else
+        F.raiseError[Unit](
+          AppCannotBeDeletedByWorkspaceIdException(workspaceId, appName, appResult.app.status, ctx.traceId)
+        )
+
+    // Get the disk to delete if specified
+    diskOpt = if (deleteDisk) appResult.app.appResources.disk.map(_.id) else None
+
+    _ <-
+      if (appResult.app.status == AppStatus.Error) {
+        for {
+          _ <- authProvider.notifyResourceDeletedV2(appResult.app.samResourceId,
```

Comment:
Yeah we call `notifyResourceDeletedV2` here, but only if the status is `Error`.. otherwise the Sam deletion will happen in Back Leo after polling for deletion is done.